### PR TITLE
Refactor get_filesystem to use native fsspec API

### LINKED ISSUE #11674: Refactor get_filesystem to use native fsspec API support
## Proposed refactor

Replace the ad-hoc logic in get_filesystem of [utilities/cloud_io.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/42fddc371af0c007b06d25c35059d84a116b3df0/pytorch_lightning/utilities/cloud_io.py#L48-L53) with the [native fsspec API](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.core.url_to_fs)

### Motivation

Use native fsspec API compared to custom logic

### Pitch

```
def get_filesystem(path: Union[str, Path]) -> AbstractFileSystem:
    fs, _ = url_to_fs(str(path))
    return fs
 ```

cc @borda @justusschock @awaelchli @akihironitta @rohitgr7

### PR DESCRIPTION
**Pull request by DuYicong515**
## What does this PR do?

Fixes #11674 

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [n/a] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by carmocca — APPROVED** [review]
Nice

File: pytorch_lightning/utilities/cloud_io.py [hunk 1]
```diff
@@ -45,12 +46,8 @@ def load(
 
 
 def get_filesystem(path: Union[str, Path]) -> AbstractFileSystem:
```
Comment (ananthsub) [inline pytorch_lightning/utilities/cloud_io.py:48]: - update the typehint to _PATH from here: https://github.com/PyTorchLightning/pytorch-lightning/blob/ec1379da2c9388b144adcb4b4cc751839f085f65/pytorch_lightning/utilities/types.py#L37
as we use this across the codebase

- `url_to_fs` supports **kwargs being passed in. https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.core.url_to_fs

i think we should add **kwargs support to get_filesystem which are threaded through to url_to_fs as well

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/utilities/cloud_io.py (hunk 1 above)
Comment (DuYicong515) [reply pytorch_lightning/utilities/cloud_io.py:48]: Sounds good!

File: pytorch_lightning/utilities/cloud_io.py [hunk 2]
```diff
@@ -19,7 +19,8 @@
 
 import fsspec
 import torch
-from fsspec.implementations.local import AbstractFileSystem, LocalFileSystem
+from fsspec.core import url_to_fs
```
Comment (rohitgr7) [inline pytorch_lightning/utilities/cloud_io.py:22]: is there any specific version after which this function is available in `fsspec`? If so, we might need to update the minimum requirements: https://github.com/PyTorchLightning/pytorch-lightning/blob/ec1379da2c9388b144adcb4b4cc751839f085f65/requirements.txt#L7

else it's fine.

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/utilities/cloud_io.py (hunk 2 above)
Comment (ananthsub) [reply pytorch_lightning/utilities/cloud_io.py:22]: it's been available for at least a year, but hadn't been publicly documented until this: https://github.com/fsspec/filesystem_spec/issues/752

Comment (DuYicong515) [reply pytorch_lightning/utilities/cloud_io.py:22]: that's a great point! It's available after [2021.09.0](https://github.com/fsspec/filesystem_spec/releases/tag/2021.09.0)
I'll update it.
